client/src/api/v2/camera.js

```diff
@@ -51,6 +51,65 @@ export function useOverlays(device) {
   return { overlays, setActiveOverlay };
 }
 
+/**
+ * @typedef {object} CameraMessageState
+ * @property {string} message The message currently entered by the user.
+ * @property {boolean} received Specifies whether a sent message has been
+ *                              acknowledged by the server in the last 5
+ *                              seconds.
+ * @property {boolean} sending Specifies whether we are waiting for an
+ *                             acknowledgement for the most recent message.
+ */
+
+/**
+ * @typedef {object} CameraMessageHook
+ * @property {CameraMessageState} state Current state of the messages.
+ * @property {function(string)} setMessage Set the message.
+ * @property {function(void)} sendMessage Send the message to the rider.
+ */
+
+/**
+ * Handle storing, sending and receiving acknowledgement of messages to the
+ * camera overlay.
+ *
+ * @returns {CameraMessageHook} Hook
+ */
+export function useMessageState() {
+  const [message, setMessage] = useState('');
+  const [received, setReceived] = useState(false);
+  const [sending, setSending] = useState(false);
+  const receivedTimeout = useRef(null);
+
+  const receivedCallback = useCallback(() => {
+    clearTimeout(receivedTimeout.current);
+    setMessage('');
+    setReceived(true);
+    setSending(false);
+
+    // Remove the "received" status after 5 seconds
+    receivedTimeout.current = setTimeout(() => {
+      setReceived(false);
+    }, 5000);
+  }, []);
+
+  useChannel('send-message-received', receivedCallback);
+
+  return { message, received, sending, setMessage, setSending };
+}
+
+/**
+ * Sends a message to be shown on the camera overlay.
+ *
+ * @param {string} message The message to be sent
+ * @returns {boolean} True only if a message has been sent.
+ */
+export function sendMessage(message) {
```

Comment:
@angustrau thoughts on moving `sendMessage` into the `useMessageState` hook and exposing it there? That way, we can keep all logic in sending/receiving inside the custom hook.

Reply:
I actually had it like that originally and then moved it out, unsure if it was what Angus wanted. See https://github.com/monash-human-power/dashboard/pull/67#discussion_r479702334